### PR TITLE
Duplex stt eartts v2

### PR DESCRIPTION
**Pull request by vtrinh-nvidia**
# What does this PR do ?

Save the agent’s speech as a separate audio  @kevinhu-nv @zhehuaic

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/speechlm2/parts/metrics/results_logger.py [hunk 1]
```diff
@@ -81,9 +97,19 @@ def merge_and_save_audio(
                 ],
                 dim=0,
             ).squeeze()
-
+            # Optionally save padded single-channel tracks
+            if agent_out_audio_path is not None:
+                # torchaudio.save(agent_out_audio_path, pred_audio_padded.squeeze().unsqueeze(0).detach().cpu(), pred_audio_sr)
```
Comment (kevinhu-nv) [inline nemo/collections/speechlm2/parts/metrics/results_logger.py]: Can you remove comments?

File: nemo/collections/speechlm2/parts/metrics/results_logger.py [hunk 2]
```diff
@@ -81,9 +97,19 @@ def merge_and_save_audio(
...
+                sf.write(agent_out_audio_path, pred_audio_padded.squeeze().unsqueeze(0).detach().cpu().numpy().astype('float32').T, pred_audio_sr)
+            if user_out_audio_path is not None:
+                # torchaudio.save(user_out_audio_path, user_audio_padded.squeeze().unsqueeze(0).detach().cpu(), pred_audio_sr)
```
Comment (kevinhu-nv) [inline nemo/collections/speechlm2/parts/metrics/results_logger.py]: ditto

File: nemo/collections/speechlm2/parts/metrics/results_logger.py (hunk 1 above)
Comment (vtrinh-nvidia) [reply nemo/collections/speechlm2/parts/metrics/results_logger.py]: Thanks @kevinhu-nv. I've removed comments in [cb357e7]

File: nemo/collections/speechlm2/parts/metrics/results_logger.py [hunk 3]
```diff
@@ -81,9 +97,16 @@ def merge_and_save_audio(
...
-
+            if agent_out_audio_path is not None:
+                sf.write(agent_out_audio_path, pred_audio_padded.squeeze().unsqueeze(0).detach().cpu().numpy().astype('float32').T, pred_audio_sr)
+            if user_out_audio_path is not None:
+                sf.write(user_out_audio_path, user_audio_padded.squeeze().unsqueeze(0).detach().cpu().numpy().astype('float32').T, pred_audio_sr)
         else:
             combined_wav = pred_audio.unsqueeze(0).detach().cpu()
+            # Save agent-only if requested even when user is absent
+            if agent_out_audio_path is not None:
+                # torchaudio.save(agent_out_audio_path, pred_audio.squeeze().unsqueeze(0).detach().cpu(), pred_audio_sr)
```
Comment (kevinhu-nv) [inline nemo/collections/speechlm2/parts/metrics/results_logger.py]: remove comments here as well

Comment (vtrinh-nvidia) [reply nemo/collections/speechlm2/parts/metrics/results_logger.py]: done !
